app/src/main/java/org/dhis2/data/forms/dataentry/SearchTEIRepositoryImpl.kt

```diff
@@ -25,20 +28,28 @@ class SearchTEIRepositoryImpl(
         val orgUnitScope = attribute.orgUnitScope() ?: false
 
         if (isUnique && !orgUnitScope) {
-            val teiList = d2.trackedEntityModule().trackedEntityInstanceQuery().onlineOnly()
-                .allowOnlineCache()
-                .eq(true)
-                .byOrgUnitMode()
-                .eq(OrganisationUnitMode.ACCESSIBLE)
-                .byProgram()
-                .eq(programUid)
-                .byAttribute(attribute.uid()).eq(value).blockingGet()
+            try {
+                val teiList = d2.trackedEntityModule().trackedEntityInstanceQuery().onlineOnly()
+                    .allowOnlineCache()
+                    .eq(true)
+                    .byOrgUnitMode()
+                    .eq(OrganisationUnitMode.ACCESSIBLE)
+                    .byProgram()
+                    .eq(programUid)
+                    .byAttribute(attribute.uid()).eq(value).blockingGet()
 
-            if (teiList.isNullOrEmpty()) {
+                if (teiList.isNullOrEmpty()) {
+                    return true
+                }
+
+                return teiList.none { it.uid() != teiUid }
+            } catch (e: Exception) {
+                crashcontroller.addBreadCrumb(
```

Comment:
maybe here que could also get D2Error?. Following [SDK documentation](https://docs.dhis2.org/en/develop/developing-with-the-android-sdk/error-management.html): 
`try {
    d2.userModule().blockingLogIn(username, password, url);
} catch (Exception e) {
    if (e.getCause() instanceof D2Error) {
        D2Error d2Error = (D2Error) e.getCause();
        Log.e("LOGIN", d2Error.errorComponent() + " " + d2Error.httpErrorCode() + " " + d2Error.errorCode());
    }
}`

Reply:
sounds good !